### PR TITLE
fix: fixes TextBox cursor color for Android 29+

### LINKED ISSUE #6240: Android Migrate off mCursorDrawableRes (setting cursor color on TextView) as throws error in api 29 and higher
## Current behavior
an exception thrown on access:
`Accessing hidden field Landroid/widget/TextView;->mCursorDrawableRes:I (greylist-max-p, reflection, denied)
**Java.Lang.NoSuchFieldException:** 'No field mCursorDrawableRes in class Landroid/widget/TextView; (declaration of 'android.widget.TextView' appears in /system/framework/framework.jar!classes3.dex)'`

## Summary
So https://github.com/unoplatform/uno/blob/14c3ae67d3f52334394afa468dcf734d9cff985c/src/Uno.UI/UI/Xaml/Controls/TextBox/TextBoxView.Android.cs#L139  in EditTextCursorColorChanger accesses the mCursorDrawableRes field which is not supported.  It was graylisted < 29 and per 29 it is blacklisted: https://developer.android.com/about/versions/10/non-sdk-q

This is referenced in https://github.com/unoplatform/uno/issues/5491 however the PR does not remove the access to this field rather more error handling.

Officially I think the recommendation is: https://developer.android.com/reference/android/widget/TextView#setTextCursorDrawable(int)  it was only added in v29 however I understand that is now the minimum version for uno anyway so could probably just swap over to using it exclusively.

### PR DESCRIPTION
**Pull request by Hudney**
GitHub Issue: closes #6240

## PR Type
- Fix

## What is the current behavior?
Error changing cursor color in API 29+

## What is the new behavior?
Fix cursor color change for Android 29+

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [x] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [x] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

# Before 
Android 10

# After
Android 11

### CODE REVIEW

File: src/Uno.UI/UI/Xaml/Controls/TextBox/TextBoxView.Android.cs [hunk 1]
```diff
@@ -146,7 +149,7 @@ private static void PrepareFields(Context? context)
 				{
 					_cursorDrawableField = _editorField.Get(editText)?.Class.GetDeclaredField("mCursorDrawable");
 				}
-				else
+				else if((int)Build.VERSION.SdkInt < 29)
 				{
 					// set differently in Android P (API 28) and higher
```
Comment (Youssef1313) [inline src/Uno.UI/UI/Xaml/Controls/TextBox/TextBoxView.Android.cs:154]: I think this comment is no longer correct. This if branch is reachable only for API level 28 exactly.

File: src/Uno.UI/UI/Xaml/Controls/TextBox/TextBoxView.Android.cs [hunk 2]
```diff
@@ -167,51 +170,66 @@ public static void SetCursorColor(EditText editText, Color color)
...
 						return;
 					}
 
-					var mCursorDrawableRes = _cursorDrawableResField.GetInt(editText);
-					var editor = _editorField.Get(editText);
-
-#if __ANDROID_28__
+					if ((int)Build.VERSION.SdkInt >= 29)
+					{
 #pragma warning disable 618 // SetColorFilter is deprecated
```
Comment (jeromelaban) [inline src/Uno.UI/UI/Xaml/Controls/TextBox/TextBoxView.Android.cs:181]: Have you found out what the replacement for SetColorFilter is from the android documentation ?

File: src/Uno.UI/UI/Xaml/Controls/TextBox/TextBoxView.Android.cs (hunk 1 above)
Comment (teras) [reply src/Uno.UI/UI/Xaml/Controls/TextBox/TextBoxView.Android.cs:154]: That's true. We'll update the `if` case and add a comment what happens with APIs 29+

File: src/Uno.UI/UI/Xaml/Controls/TextBox/TextBoxView.Android.cs (hunk 2 above)
Comment (teras) [reply src/Uno.UI/UI/Xaml/Controls/TextBox/TextBoxView.Android.cs:181]: We'll  use the proposed BlendModeColorFilter which is the one proposed by Android documentation instead.
https://developer.android.com/reference/android/graphics/BlendModeColorFilter

**Comment by jeromelaban** [comment]
Thanks for the update. Make sure to squash and rebase your PR (so the conventional commits are valid).

File: src/Uno.UI/UI/Xaml/Controls/TextBox/TextBoxView.Android.cs (hunk 1 above)
Comment (Hudney) [reply src/Uno.UI/UI/Xaml/Controls/TextBox/TextBoxView.Android.cs]: Done.

**Comment by jeromelaban** [comment]
This is looking great! Can you make sure to rebase your commit on master? It looks like there are some other unrelated commits.

**Pull request closed without merging**